pkg/reconciler/source/config_watcher.go

```diff
@@ -20,143 +20,263 @@ import (
 	"context"
 
 	"go.uber.org/zap"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/eventing/pkg/logging"
 	"knative.dev/pkg/configmap"
-	pkgLogging "knative.dev/pkg/logging"
+	pkglogging "knative.dev/pkg/logging"
 	"knative.dev/pkg/metrics"
 	tracingconfig "knative.dev/pkg/tracing/config"
+)
 
-	"knative.dev/eventing/pkg/logging"
+const (
+	EnvLoggingCfg = "K_LOGGING_CONFIG"
+	EnvMetricsCfg = "K_METRICS_CONFIG"
+	EnvTracingCfg = "K_TRACING_CONFIG"
 )
 
+type ConfigAccessor interface {
+	ToEnvVars() []corev1.EnvVar
+	LoggingConfigEnvVar() corev1.EnvVar
```

Comment:
My idea was more to access not to the env var, but to the materialized configs eg `*pkglogging.Config`

Reply:
OK, now I get it.

Reply:
Done in [130a0e4](https://github.com/knative/eventing/compare/e640e85fdad4a388ca22ede8bc0861c6fd904340..130a0e4a90efeb812de74391d0a4cfc2e37e5d3d)